### DISCUSSION TITLE
modularize existing modules

### DISCUSSION THREAD

**Discussion by erikLundstedt** [SystemCrafters/crafted-emacs#243] [Ideas]
for example, crafted-ui installs doom-modeline and some other packages that not everyone will want
maybe it would be good to make a crafted-modeline module that allows the user to install whatever modeline package they want, see [uncrafted-modeline.el](https://gitlab.com/Erik.Lundstedt/dotconfigfiles/-/blob/9da3be70b4750825d9704f1a3256e34eec2335df/.config/crafted-emacs/custom-modules/uncrafted-modeline.el) from my config,for an example(ugly code, but it works)(only installs the wanted package)

another module that could easily be made is a crafted-helpful.el for the helpful.el(?) config currently in crafted-ui

implementing this would allow for less code in the users config (like disabling hooks created in a module) and faster install as an unused package just takes extra time when doing package management

**Comment by jeffbowman** [comment]
Given the recent issues surrounding `doom-modeline`, this seems like a reasonable idea.  My concern would be where to draw the line, because it starts to become 1 crafted module per package, which is far too granular (in my opinion), but then trying to decide which packages to group together vs which ones to segregate becomes a situation where few really win. 

The principle of `reversible decision` is there to handle this kind of situation.  I don't use `doom-modeline`, so I turn it off in my configuration - however, that doesn't prevent it from being installed. Same with `helpful`.  From the `crafted-completion` module I prefer to add prescient (and I used to prefer selectrum - but it is now deprecated in favor of vertico, and I don't like how that one works, so back to icomplete-vertical-mode (now built-in) for me).
